cub/cub/block/radix_rank_sort_operations.cuh

```diff
@@ -73,9 +74,14 @@ CUB_NAMESPACE_BEGIN
     and only one of them is used, the sorting works correctly. For double, the
     same applies, but with 64-bit patterns.
 */
-template <typename KeyT, Category TypeCategory = Traits<KeyT>::CATEGORY>
+template <typename KeyT, bool IsFP = ::cuda::is_floating_point<KeyT>::value>
```

Comment:
should this (and others) be using `_CCCL_TRAIT`?

Reply:
Note that almost all other sues are `is_primitive` which is defined through a struct, so there is no real benefit